File: src/main/java/Piruru/cards/DiscardAnyAmountAndGainEnergy.java
package Piruru.cards;

import Piruru.abstracts.PiruruCard;
import Piruru.actions.DiscardAnyAmountAction;
import com.megacrit.cardcrawl.actions.common.GainEnergyAction;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;



public class DiscardAnyAmountAndGainEnergy extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(DiscardAnyAmountAndGainEnergy.class.getSimpleName()));
    private static final int COST = 1;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.RARE;
    private static final CardTarget TARGET = CardTarget.SELF;
    private static final int COST_UP = 0;
    private static final int ENERGY_GAIN = 1;

    public DiscardAnyAmountAndGainEnergy() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST_UP);
        baseMagicNumber = magicNumber = ENERGY_GAIN;
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        act(new DiscardAnyAmountAction(list -> list.forEach(
                c -> act(new GainEnergyAction(magicNumber))
        )));
    }
}

File: src/main/kotlin/Cirno/cards/DiscardAnyAmountAndGainEnergy.kt
package Cirno.cards

import Cirno.Cirno.Statics.makeID
import Cirno.abstracts.CirnoCard
import Cirno.actions.DiscardAnyAmountAction
import com.megacrit.cardcrawl.actions.common.GainEnergyAction
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.monsters.AbstractMonster


class DiscardAnyAmountAndGainEnergy : CirnoCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, 0, 0, COST_UP) {
    init {
        magicNumber = ENERGY_GAIN
        baseMagicNumber = magicNumber
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        act(DiscardAnyAmountAction { list -> list.forEach { c -> act(GainEnergyAction(magicNumber)) } })
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(DiscardAnyAmountAndGainEnergy::class.java.simpleName))
        private val COST = 1
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.RARE
        private val TARGET = AbstractCard.CardTarget.SELF
        private val COST_UP = 0
        private val ENERGY_GAIN = 1
    }
}
